libs/device/feature/src/lib/api-entities/api-entities.service.ts

```diff
@@ -43,34 +55,149 @@ export class APIEntitiesService {
     return Result.success(response.data)
   }
 
-  @IpcEvent(APIEntitiesServiceEvents.EntityConfig)
-  public async getEntityConfiguration({
+  @IpcEvent(APIEntitiesServiceEvents.EntitiesConfig)
+  public async getEntitiesConfiguration({
     entityType,
     deviceId,
   }: {
     entityType: string
     deviceId?: DeviceId
-  }): Promise<ResultObject<EntityConfig>> {
+  }): Promise<ResultObject<EntitiesConfig>> {
     const device = this.getDevice(deviceId)
     if (!device) {
       return Result.failed(new AppError(GeneralError.NoDevice, ""))
     }
 
     const response = await device.request({
-      endpoint: "ENTITY_CONFIGURATION",
+      endpoint: "ENTITIES_CONFIGURATION",
       method: "GET",
       body: {
-        type: entityType,
+        entityType,
       },
     })
     if (!response.ok) {
       return this.handleError(response.error.type)
     }
 
-    const apiConfig = entityConfigValidator.safeParse(response.data.body)
+    const apiConfig = entitiesConfigValidator.safeParse(response.data.body)
     if (!apiConfig.success) {
       return this.handleError(response.data.status)
     }
     return this.handleSuccess(apiConfig)
   }
+
+  @IpcEvent(APIEntitiesServiceEvents.EntitiesMetadata)
+  public async getEntitiesMetadata({
+    entityType,
+    deviceId,
+  }: {
+    entityType: string
+    deviceId?: DeviceId
+  }): Promise<ResultObject<EntitiesMetadata>> {
+    const device = this.getDevice(deviceId)
+    if (!device) {
+      return Result.failed(new AppError(GeneralError.NoDevice, ""))
+    }
+
+    const response = await device.request({
+      endpoint: "ENTITIES_METADATA",
+      method: "GET",
+      body: {
+        entityType,
+      },
+    })
+    if (!response.ok) {
+      return this.handleError(response.error.type)
+    }
+
+    const metadata = entitiesMetadataValidator.safeParse(response.data.body)
+    if (!metadata.success) {
+      return this.handleError(response.data.status)
+    }
+    return this.handleSuccess(metadata)
+  }
+
+  @IpcEvent(APIEntitiesServiceEvents.EntitiesDataGet)
+  public async getEntitiesData({
+    entityType,
+    entityId,
+    responseType,
+    deviceId,
+  }: {
+    entityType: string
+    responseType: "json" | "file"
+    entityId?: string | number
```

Comment:
Wouldn't it be better to only use a string here? I see that in other places and in the documentation we consistently operate with strings.

Reply:
We are, but it's generic and numeric IDs are also a valid way of indexing items.